addons/sale/models/sale_order.py

```diff
@@ -697,11 +697,9 @@ def _fetch_duplicate_orders(self):
                  AND sale_order.id != duplicate_order.id
                  AND duplicate_order.state != 'cancel'
                  AND sale_order.partner_id = duplicate_order.partner_id
-                 AND sale_order.date_order = duplicate_order.date_order
```

Comment:
Your compute depends and `used_fields` still refer to `date_order`. Not sure if it's supposed to be removed here or from the other references, but it should probably be consistent